apps/aecore/lib/aecore/wallet/worker.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Aecore.Wallet.Worker do
+  @defmodule """
+  Module for handling the creation of a Wallet file
+  """
+
+  use GenServer
+
+  alias Aewallet.Wallet
+
+  @typedoc "Options for network"
+  @type opts :: :mainnet | :testnet
+
+  @aewallet_dir Application.get_env(:aecore, :aewallet)[:path]
+  @aewallet_pass Application.get_env(:aecore, :aewallet)[:pass]
+
+  ## Client API
+
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{path: @aewallet_dir, pubkey: nil}, name: __MODULE__)
+  end
+
+  def init(%{path: path} = state) do
+    :ok = path
+    |> File.mkdir()
+    |> has_wallet(path)
+
+    {:ok, state}
+  end
+
+  @spec get_public_key(String.t()) :: binary()
+  def get_public_key(password) do
+    GenServer.call(__MODULE__, {:get_pub_key, {password, :mainnet}})
+  end
+
+  @spec get_public_key(String.t(), opts()) :: binary()
+  def get_public_key(password, network) do
+    GenServer.call(__MODULE__, {:get_pub_key, {password, network}})
+  end
+
+  @spec get_private_key(String.t()) :: binary()
+  def get_private_key(password) do
+    GenServer.call(__MODULE__, {:get_priv_key, {password, :mainnet}})
+  end
+
+  @spec get_private_key(String.t(), opts()) :: binary()
+  def get_private_key(password, network) do
+    GenServer.call(__MODULE__, {:get_priv_key, {password, network}})
+  end
+
+  ## Server Callbacks
+
+  def handle_call({:get_pub_key, {password, network}}, _from, %{pubkey: nil} = state) do
```

Comment:
I think this genserver could also handle "password" and "network" parameters internally. (store them, probably load them from config)

Reply:
Having them stored initialy actually loses the purpose of having them in the first place. In my opinion a password should not be stored anywhere and the network should be able to change on the fly.

Reply:
- I think the network should be part of some config, becouse changing the network requires preaty much to restart everything (different database for blocks, different genesis block, different peers). Also even if we would want to change it on the fly it still would be nice if the information would be stored in the GenServer, this way code that wants to sign something doesn't have to pass the information about network every time. 
- By storing password I mean storing in the state of GenServer in RAM.

Reply:
May be we can let the user to decide if he/she wants to keep the password into the memory or to type it every time.